module-services/service-desktop/ServiceDesktop.cpp

```diff
@@ -53,6 +54,38 @@ auto ServiceDesktop::getSerialNumber() const -> std::string
     return settings->getValue(std::string("factory_data/serial"), settings::SettingsScope::Global);
 }
 
+auto ServiceDesktop::generateDeviceUniqueId() -> void
```

Comment:
Moved generation of Device Unique Id outside on-boarding, because it won't work after OS update on the already on-boarded device.